src/parsers/onlinerParser/index.ts

```diff
@@ -0,0 +1,103 @@
+import axios from 'axios';
+import FormData = require('form-data');
+import * as _ from 'lodash';
+import fetch from 'node-fetch';
+import puppeteer = require('puppeteer');
+import { codeErrors } from '../../config/config';
+import { ParserError } from '../../utils/errors/';
+
+const getPage = async (url: string) => {
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+  await page.goto(url, { waitUntil: 'networkidle' });
+  const content = page.content();
+
+  return content;
+};
+
+export const getMarks = async () => {
+  const res = await getPage(`https://ab.onliner.by/`);
+  // match manufactures
+  const modelsData = res.match(/var\s+Manufactures \= \[.*\]/g);
+  const arrayOfMarks = modelsData[0].match(/[{][^\}]*[}]/g);
+  for (let indexOFCar = 0; indexOFCar < arrayOfMarks.length; indexOFCar++) {
+    arrayOfMarks[indexOFCar] = JSON.parse(arrayOfMarks[indexOFCar]);
+  }
+  return arrayOfMarks;
+};
+
+export const getModels = async () => {
+  const res = await getPage(`https://ab.onliner.by/`);
+  // match manufactures models
+  const modelsData: any = res.match(/var\s+ManufacturesModel \= \{.*\}/g);
+  const models: any = modelsData[0].match(/\{.*\}/g);
+  return JSON.parse(models[0]);
+};
+
+export const getBodyTypes = async () => {
+  let res = await getPage(`https://ab.onliner.by/`);
+  const bodyTypes: any = [];
+  // remove all spaces for more comfort parsing
+  res = res.replace(/ /g, '');
+  // match all structure constructions of body types select
+  const listOfTypes: any = res.match(
+    /\<inputtype="checkbox"name="body_type\[\]"class="f-cb"value="\d+"\>[\r\n]*.*/g
```

Comment:
We discussed that we need to use cheerio for this. Why are we still using the RegEx?

Reply:
It is still  RegEx here, cause we created new task for cherrio and assign it to Liza